File: leetcode/twoSum.js
var twoSum = function (nums, target) {
  const tracker = {};
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] in tracker) {
    } else {
      tracker[nums[i]] = i;
    }
  }
  let difference;
  for (let i = 0; i < nums.length; i++) {
    difference = target - nums[i];
    if (difference in tracker) {
      if (i !== tracker[difference]) {
        return [i, tracker[difference]];
      }
    }
  }
};

File: leetcode/two_sum.py
def two_sum(nums, target):
    tracker = {}
    difference = 0
    for x in range(len(nums)):
        difference = target - nums[x]
        if difference in tracker:
            return [tracker[difference], x]
        if nums[x] in tracker:
            print("do nothing")
        else:
            tracker[nums[x]] = x


list = [3, 3]
print(two_sum(list, 6))
